adhocracy4/comments_async/static/comments_async/comment_form.jsx

```diff
@@ -92,8 +92,8 @@ export default class CommentForm extends React.Component {
     }
     this.props.onCommentSubmit(data, this.props.parentIndex).then(() => {
       this.clearForm()
-      if (this.props.useTermsOfUse) {
-        this.setState({ agreedTermsOfUse: this.state.checkedTermsOfUse })
+      if (this.props.useTermsOfUse && !this.state.agreedTermsOfUse && this.state.checkedTermsOfUse) {
+        this.setState({ agreedTermsOfUse: true })
```

Comment:
Well the `agreedTermsOfUse` comes from the api and reflects what is stored in the database. If we would refetch the comments after posting, we would not need that as a state, I think, and could only use props. But since we dont, we change the state after the agremment has been sent manually.
And the `checkedTermsOfUse` is just if the checkbox is checked or not.
Does that make sense?